### PR TITLE
Fronts: Media card play icon

### PR DESCRIPTION
**Pull request by OllysCoding**
## What does this change?

Adds a 'play icon' to cards configured with 'showMainVideo' in fronts tool:

image

This uses designs provided by @HarryFischer, which slightly differ from the existing frontend styling

## Screenshots

| Desktop      | Mobile      |
|-------------|------------|
| ![desktop][] | ![mobile][] |

[desktop]: https://user-images.githubusercontent.com/9575458/214546035-a43a2763-a5c4-4361-bb57-3964a0d66c91.png

[mobile]: https://user-images.githubusercontent.com/9575458/214546058-dab24c88-6630-45ba-9a81-7e59c1bd5935.png

| Frontend      | DCR      |
|-------------|------------|
| ![frontend][] | ![dcr][] |
| ![frontend2][] | ![dcr2][] |

[frontend]: https://user-images.githubusercontent.com/9575458/214091141-179ff018-6453-401f-8804-5221c0b15d90.png
[dcr]: https://user-images.githubusercontent.com/9575458/214546799-6a7e3aa4-abcd-4f55-99e3-0b28a7613b11.png

[frontend2]: https://user-images.githubusercontent.com/9575458/214091583-4bf428f3-49f1-4190-81d0-e40a7346aa9a.png
[dcr2]: https://user-images.githubusercontent.com/9575458/214547210-a1333367-6e85-4da6-86d0-83f0d402c81b.png

### CODE REVIEW

File: dotcom-rendering/src/model/enhanceCards.ts [hunk 1]
```diff
@@ -86,8 +86,8 @@ const enhanceSupportingContent = (
 
 		return {
 			format: presentationFormat,
-			headline: subLink.header?.headline || '',
-			url: subLink.properties.href || subLink.header?.url,
+			headline: subLink.header?.headline ?? '',
+			url: subLink.properties.href ?? subLink.header?.url,
```
Comment (OllysCoding) [inline dotcom-rendering/src/model/enhanceCards.ts:90]: Just sorting out some ESLint warnings :)

File: dotcom-rendering/src/web/components/Card/components/PlayIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	imageSize: ImageSizeType,
+	imagePosition: ImagePositionType,
+) => {
+	if (imageSize === 'jumbo') return 60;
+	else if (
+		(imagePosition === 'left' || imagePosition === 'right') &&
+		imageSize === 'small'
+	)
+		return 24;
+	else return 40;
```
Comment (jamesgorrie) [inline dotcom-rendering/src/web/components/Card/components/PlayIcon.tsx:38]: This is a little fiddly to read from my perspective, maybe some `{}` would help?

File: dotcom-rendering/src/model/article-schema.json [hunk 3]
```diff
@@ -4317,6 +4317,9 @@
                 },
                 "showQuotedHeadline": {
                     "type": "boolean"
+                },
+                "showMainVideo": {
```
Comment (jamesgorrie) [inline dotcom-rendering/src/model/article-schema.json:4321]: I can't see where this is set or used - maybe a copy/paste error?

**Review by jamesgorrie — COMMENTED** [review]
Great! Just a question on whether we should have `showMainVideo` on the article schema.

File: dotcom-rendering/src/model/article-schema.json (hunk 3 above)
Comment (OllysCoding) [reply dotcom-rendering/src/model/article-schema.json:4321]: It's passed to the `ImageWrapper` component, where we decide whether or not to include the the `PlayIcon` component

File: dotcom-rendering/src/web/components/Card/components/PlayIcon.tsx (hunk 2 above)
Comment (OllysCoding) [reply dotcom-rendering/src/web/components/Card/components/PlayIcon.tsx]: I think this is improved in the latest revision :)

**Review by HarryFischer — APPROVED** [review]
Nice - thanks Olly!

File: dotcom-rendering/src/web/lib/cardWrappers.tsx [hunk 4]
```diff
@@ -435,7 +394,7 @@ export const Card50Media50 = ({
 			containerPalette={containerPalette}
 			headlineSize="large"
 			headlineSizeOnMobile="large"
-			imageSize="small"
+			imageSize="medium"
```
Comment (OllysCoding) [inline dotcom-rendering/src/web/lib/cardWrappers.tsx:397]: Being more explicit about these image sizes for the top/bottom images allows us to correctly size the play icon in the card

Comment (OllysCoding) [reply dotcom-rendering/src/web/lib/cardWrappers.tsx:397]: Until now, the image size hasn't really mattered with the image position is top/bottom

**Review by ioannakok — APPROVED** [review]
Looks great!

File: dotcom-rendering/src/web/components/Card/components/PlayIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
+import { css } from '@emotion/react';
+import { brandAlt, from } from '@guardian/source-foundations';
+import { SvgMediaControlsPlay } from '@guardian/source-react-components';
+
+type PlayButtonSize = 'small' | 'medium' | 'large' | 'xlarge';
```
[the remark this answers is not in the document]
Comment (OllysCoding) [reply dotcom-rendering/src/web/components/Card/components/PlayIcon.tsx:5]: As it's very self contained (just this one file), I think it'd make sense to keep it here for the time being, I don't anticipate that there would be a reason we'd need to use it outside this context!

**Pull request merged**